File: service/api/schema/company.py
from marshmallow import Schema, fields

class CompanySchema(Schema):
    company_id = fields.Int(required=True)
    name = fields.Str(required=True)
    address = fields.Str(required=True)
    latitude = fields.Float(required=True)
    longitude = fields.Float(required=True)

File: app/src/models/Company.ts
export interface Company {
    company_id: number;
    name: string;
    address: string;
    latitude: number;
    longitude: number;
    locations: Location[];
}

export interface Location {
    location_id: number;
    company_id: number;
    name: string;
    address: string;
    latitude: number;
    longitude: number;
}
